lib/cmsis_rtos_v1/Kconfig

```diff
@@ -5,6 +5,7 @@ config CMSIS_RTOS_V1
 	bool "CMSIS RTOS v1 API"
 	depends on THREAD_CUSTOM_DATA
 	depends on POLL
+	select LEGACY_TIMEOUT_API
```

Comment:
we should probably file an enhancement GH ticket that cmsis v1/v2 and POSIX currently forces the legacy API

edit: net buffers and console as well. how much effort is it to convert these?

Reply:
Not a lot, really.  But it involves a little bit of design decision (i.e. does the subsystem want to expose k_timeout_t in its own API or stick with milliseconds and convert, etc..).  One of the sticking points last time was adding changes to subsystems without maintainer approval, so this one limits itself to subsystems that port with only trivial changes.

Reply:
OK. I think  we should have GH tickets just so we can track it.